storage/storage.go

```diff
@@ -348,3 +395,205 @@ func EncodeName(name string) string {
 	}
 	return result
 }
+
+func SeparateMod(ctx context.Context, body []byte, modID, name string, versionID string, modVersion string) (bool, string) {
+	//read combined file
+	zipReader, err := zip.NewReader(bytes.NewReader(body), int64(len(body)))
+
+	if err != nil {
+		return false, ""
+	}
+
+	// Create a buffer to write our archive to.
+	log.Ctx(ctx).Info().Msg("Beginning Splitting Operations")
+	bufLinuxServer := new(bytes.Buffer)
+	bufWin64Server := new(bytes.Buffer)
+	bufWin64Client := new(bytes.Buffer)
+
+	// Create a new zip archive.
+	zipWriterLinuxServer := zip.NewWriter(bufLinuxServer)
+	zipWriterWin64Server := zip.NewWriter(bufWin64Server)
+	zipWriterWin64Client := zip.NewWriter(bufWin64Client)
+
+	var LinuxServer = false
+	var Win64Server = false
+	var Win64Client = false
+
+	cleanName := cleanModName(name)
+
+	// Add some files to the archive.
+	for _, file := range zipReader.File {
+		if strings.HasPrefix(file.Name, ".pdb") || strings.HasPrefix(file.Name, ".debug") || strings.Contains(file.Name, modVersion) {
+			continue
+		}
+
+		if strings.Contains(file.Name, "LinuxServer") {
+			err = WriteZipFile(file, "LinuxServer", zipWriterLinuxServer)
+
+			if err != nil {
+				fmt.Println(err)
+				return false, ""
+			}
+
+			LinuxServer = true
+		}
+
+		if strings.Contains(file.Name, "WindowsServer") {
+			err = WriteZipFile(file, "WindowsServer", zipWriterWin64Server)
+
+			if err != nil {
+				fmt.Println(err)
+				return false, ""
+			}
+
+			Win64Server = true
+		}
+
+		if strings.Contains(file.Name, "WindowsNoEditor") {
+			err = WriteZipFile(file, "WindowsNoEditor", zipWriterWin64Client)
+
+			if err != nil {
+				fmt.Println(err)
+				return false, ""
+			}
+
+			Win64Client = true
+		}
+	}
+
+	// Make sure to check the error on Close.
+	err = zipWriterLinuxServer.Close()
+	if err != nil {
+		fmt.Println(err)
+		return false, ""
+	}
+	err = zipWriterWin64Server.Close()
+	if err != nil {
+		fmt.Println(err)
```

Comment:
Please log the error using the logging library

Reply:
https://github.com/satisfactorymodding/smr-api/pull/12/commits/74cbf9e7fb628b477e7bb2459a0f641867ec9415